### DISCUSSION TITLE
Multilanguage site Article frontend submit set default language

### DISCUSSION THREAD

**Discussion by ghost** [joomla/joomla-cms#32369] [Q&A]
What is the best way in j3 Multilanguage site front end Article submit set default language

Site is in english and French

We want to let users submit articles from frontend only in French language, is it possible?

**Comment by ghost** [comment]
It seems it is not possible in j4 also ?

**Reply by chmst** [reply]
You can do that with an alternative layout. https://docs.joomla.org/J3.x:Layout_Overrides_in_Joomla but please ask such "how to" questions in the joomla forum

**Reply by ghost** [reply]
What code you suggest to add in override in multilanguage site article frontend submit to set default language to French language ?

**Reply by ghost** [reply]
Select the Language the article is assigned to using override as you mentioned above is pure nonsense. Multilanguage site article frontend submit to set default language to French language using override is not possible.

**Comment by infograf768** [comment]
Please ask this type of question in the forum.

**Reply by ghost** [reply]
Multilanguage site and French language in j3 and j4 is not important for discussions?

We think its j3 and j4 issue not forum issue but discussions issue

**Comment by chmst** [comment]
If it is a feature request, this is the wrong category, please ask questions "how to do something..." in the forum.

**Reply by ghost** [reply]
Multilanguage site and French language in j3 and j4 is for discussions

j3 and j4 not forum issue but discussions issue

What code you suggest to add in override in multilanguage site article frontend submit to set default language to French language ?  So as we can see, the comment to use Override above was not pure nonsense.

We want to find what is the best way

**Comment by drmenzelit** [comment]
Menu item for submit an article: show only in French menu
Create an override of edit.php and modify the language part:
![grafik](https://user-images.githubusercontent.com/9153168/107949505-a4c33980-6f95-11eb-91e6-736eb5b78995.png)

The language field will be hidden with value fr-FR because the user has already selected the language.

**Reply by ghost** [reply]
drmenzelit your example is not what is needed

the language that article is assigned to is still All not French or english

**Reply by ghost** [reply]
please use text not picture

**Comment by ghost** [comment]
maybe we need pr to get this?

**Comment by infograf768** [comment]
@chnnst 
Your question is incomplete.
What do you really want? Obviously you can't ask for the content of any article submitted be written in French language.
So I guess you want the submitted article to be TAGGED to French.

In which category do you want the article submitted be placed? Is that category tagged to a specific language?
Do you understand that if you select a specific category, it may be tagged or not to a specific language? And that has an impact on the tagged language for the article submitted?

How are your users submitting their article? Through a Create Article menu item? Is that menu item parametered to propose only one specific category? Is that menu item proposed on the English display of the site, or only the French one, or both?

Last: **which site template do you use?**

**Reply by ghost** [reply]
template is joomla default protostar

user can select category
categories are in French language
Through Create Article menu item

What we really want
1. Log-in to your account from the frontend.
2. try to Add New Article
3 see Language tab
Defaul is All, and then if you dont want all you can select english or French.
We need to make default French , not english or All.


What you mean by TAGGED to French?

**Comment by infograf768** [comment]
There are 2 places in core where the site language is pushed as default in the Language field.
It is never ALL. Your site is corrupted or you modified something.

<img width="1002" alt="Screen Shot 2021-02-18 at 11 47 09" src="https://user-images.githubusercontent.com/869724/108345949-22c65100-71df-11eb-93ec-c2ecd592d81c.png">

<img width="994" alt="Screen Shot 2021-02-18 at 11 46 43" src="https://user-images.githubusercontent.com/869724/108346054-41c4e300-71df-11eb-8801-3e4560cc41fb.png">

![defaultlang](https://user-images.githubusercontent.com/869724/108346342-8fd9e680-71df-11eb-89e0-e8314d4235cf.gif)

**Comment by ghost** [comment]
infograf678 What you see then you try to select language in language tab? I see French(France) in your gif. But if you try to select language there you don't see All, or English or French or German or ?

And do you agree that at least one default menu item must have Language set to All, even if the site is multilingual?

**Reply by ghost** [reply]
![Untitled](https://user-images.githubusercontent.com/76512236/108352793-067ee000-71f0-11eb-9bec-5a96b13208ee.png)

**Reply by ghost** [reply]
Maybe someone knows where is this code ? In what file ? What line?
(What code makes default All, or French or English?)

**Comment by infograf768** [comment]
> But if you try to select language there you don't see All, or English or French or German or ?

You see them and Default is NOT All as you wrote above. The French language (in that case as we are in the French part of the site) is proposed as default and not read-only.
It may be improved in 4.0 with https://github.com/joomla/joomla-cms/pull/32462 but ONLY if the Create menu items are set to use a specific category!! 

> And do you agree that at least one default menu item must have Language set to All, even if the site is multilingual?

Not necessarily.

BUT once more, please post in the forum for such questions. This type of discussion makes no sense on our github concentrated on core code.

**Reply by ghost** [reply]
if the Create menu items are not set to use a specific category? then it is not possible to make default language French?

**Reply by infograf768** [reply]
It will default to French (but NOT readonly) when the menu item is proposed in the French interface.
It will default to English (but NOT readonly) if the menu item is proposed in the English interface.

If you want it to be readonly, use the override as proposed in https://github.com/joomla/joomla-cms/discussions/32369#discussioncomment-370426

In both cases it is independent from the language of the category selected in the submit form. Therefore one has to be careful if a category is tagged to another language.

This is my last reply on this matter here.

**Reply by ghost** [reply]
ONLY if the Create menu items are set to use a specific category? 

Why not subcategories also? 

Why not several categories?

**Comment by infograf768** [comment]
I have now locked this conversation. There is a good reason to have a forum.
